tasks/init.rb

```diff
@@ -0,0 +1,88 @@
+#!/opt/puppetlabs/puppet/bin/ruby
+require 'facter'
+require 'json'
+
+params   = JSON.parse(STDIN.read)
+timeout  = params['timeout'].to_i
+message  = params['message']
+
+def async_command(cmd)
+  wait_time = 3
+  case Facter.value(:kernel)
+  when 'windows'
+    prefix = "ping 127.0.0.1 -n #{wait_time} -w 1000"
```

Comment:
Sorry to burst your bubble here.  Off by one error.

ping 127.0.0.1 -n 1 -w 1000   <--- returns almost immediately
ping 127.0.0.1 -n 2 -w 1000   <--- returns after one second.